core/shared/src/test/scala/scalaz/zio/internal/StackBoolSpec.scala

```diff
@@ -1,78 +1,61 @@
 package scalaz.zio.internal
 
-import org.specs2.Specification
+import org.scalacheck.{ Arbitrary, Gen, Prop }
+import org.specs2.{ ScalaCheck, Specification }
 
-class StackBoolSpec extends Specification {
+class StackBoolSpec extends Specification with ScalaCheck {
   def is =
     "StackBoolSpec".title ^ s2"""
         Size tracking                 $e0
-        From/to list small identity   $e1
-        From/to list large identity   $e2
-        Small push/pop example        $e3
-        Large push/pop example        $e4
-        Peek/pop identity             $e5
+        From/to list identity         $e1
+        Push/pop example              $e2
+        Peek/pop identity             $e3
     """
 
-  def e0 = {
-    val list = List.fill(100)(true)
+  import Arbitrary._
 
-    StackBool(list: _*).size must_== list.length
-  }
+  private val generator: Gen[List[Boolean]] = boolListGen(0, 400)
 
-  def e1 = {
-    val list = (1 to 200).map(_ % 2 == 0).toList
-
-    StackBool(list: _*).toList must_=== list
-  }
-
-  def e2 = {
-    val list = (1 to 400).map(_ % 2 == 0).toList
-
-    StackBool(list: _*).toList must_=== list
-  }
-
-  def e3 = {
-    val stack = StackBool()
-
-    stack.push(true)
-    stack.push(true)
-    stack.push(false)
-
-    val v1 = stack.popOrElse(true)
-    val v2 = stack.popOrElse(false)
-    val v3 = stack.popOrElse(false)
-
-    (v1 must_=== false) and
-      (v2 must_=== true) and
-      (v3 must_=== true)
-  }
+  def e0 =
+    Prop.forAll(generator) { list: List[Boolean] =>
+      StackBool(list: _*).size must_== list.length
+    }
 
-  def e4 = {
-    val stack = StackBool()
+  def e1 =
+    Prop.forAll(generator) { list: List[Boolean] =>
+      StackBool(list: _*).toList must_=== list
+    }
 
-    val list = (1 to 400).map(_ % 2 == 0).toList
+  def e2 =
+    Prop.forAll(generator) { list: List[Boolean] =>
+      val stack = StackBool()
 
-    list.foreach(stack.push(_))
+      list.foreach(stack.push(_))
 
-    list.reverse.foldLeft(true must_=== true) {
-      case (result, flag) =>
-        result and (stack.popOrElse(!flag) must_=== flag)
+      list.reverse.foldLeft(true must_=== true) {
```

Comment:
`true must_=== true` - I was looking for something like `MatchSucceed` but couldn't find.

Reply:
Not sure if this is something I should act on based on your comment? I followed the already existing pattern here. I haven't used specs2 before so I'm unaware of best practices in this case.

Reply:
No action required from my side :+1: